build/build-parse-trees.js

```diff
@@ -58,7 +58,11 @@ names.forEach(function(name) {
   });
   print('   */');
   print('  constructor(%s) {', paramNames.join(', '));
-  paramNames.forEach(printInitializer);
+  print('    super(location);');
+  if (paramNames[0] !== 'location') {
```

Comment:
I think this would be clearer if we output the `super(location)` after we check that it is `paramName[0]`